test/hotspot/jtreg/runtime/ErrorHandling/ShowRegistersOnAssertTest.java

```diff
@@ -70,13 +70,21 @@ private static void do_test(boolean do_assert, // true - assert, false - guarant
         if (show_registers_on_assert) {
             // Extract the hs_err_pid file.
             File hs_err_file = HsErrFileUtils.openHsErrFileFromOutput(output_detail);
-            Pattern[] pattern = new Pattern[] { Pattern.compile("Registers:"), null };
+            Pattern[] pattern = null;
             if (Platform.isX64()) {
-                pattern[1] = Pattern.compile("RAX=.*");
+                pattern = new Pattern[] { Pattern.compile("Registers:"), Pattern.compile("RAX=.*")};
             } else if (Platform.isX86()) {
-                pattern[1] = Pattern.compile("EAX=.*");
+                pattern = new Pattern[] { Pattern.compile("Registers:"), Pattern.compile("EAX=.*")};
             } else if (Platform.isAArch64()) {
-                pattern[1] = Pattern.compile("R0=.*");
+                pattern = new Pattern[] { Pattern.compile("Registers:"), Pattern.compile("R0=.*")};
+            } else if (Platform.isS390x()) {
+                pattern = new Pattern[] { Pattern.compile("General Purpose Registers:"),
+                                          Pattern.compile("^-{26}$"),
+                                          Pattern.compile("  r0  =.*")};
+            } else if (Platform.isPPC()) {
+                pattern = new Pattern[] { Pattern.compile("Registers:"), Pattern.compile("pc =.*")};
+            } else {
+                pattern = new Pattern[] { Pattern.compile("Registers:"), null };
```

Comment:
Doesn't `null` cause other NullPointerExceptions? Did you test it?

Reply:
I only tested on s390x and my team mate tested on PPC. Both of these platforms are okay. For rest neither I have hardware access nor I have hs_err file info. So It should give them the same result  pass/fail as it was giving before this change.

Reply:
I disagree. The code should not crash. It can easily be tested by commenting out the platform case of your choice above.

Reply:
It should work if you remove ", null".

Reply:
I agree with your point that if platform check is not defined then test will crash.

> The code should not crash.

Could you suggest any other way through which, on other platforms, test will not crash instead it will notify that information is missing and test should be updated according to their platform ? Because what I see is that the whole point of null being there was to check that "registers" info is not printed.

Maybe we can ping other maintainers here and they can verify it passes and if not then add the check with this PR.

Reply:
You can ask the maintainers. But I don't think it is important. I could live with checking for "Registers:" only.

Reply:
Sure. I will remove `null` then.